### PR TITLE
WIP: Add Deno.hostname() method

### PR DESCRIPTION
**Pull request by crabmusket**
This adds `Deno.hostname()` which gets the hostname of the current machine. E.g. the same as running `uname -n` or `hostname` on linux.

TODO:
- [ ] Add 3rd party crate from lunaryorn
- [ ] Tests, linting/formatting
- [ ] Check for `allow-env` permission

Questions:
- Should we just depend on another crate that wraps the hostname calls?
- Is there any point returning an error from `get_hostname` or should it be an `Option`?
- Should `Deno.hostname` return `string | null`, or throw an exception, to account for errors? At the moment it returns an empty string if there's an error
- Should there be a permission check? Probably `allow-env`?
- How are my string conversions?

References:
- https://github.com/lunaryorn/gethostname.rs/blob/master/src/lib.rs
- https://github.com/Aldarobot/whoami/blob/master/src/unix.rs
- https://github.com/fengcen/hostname/blob/master/src/lib.rs
- https://bitbucket.org/DaveLancaster/winutil/src/master/src/lib.rs
- https://golang.org/src/os/sys_linux.go (uses the `uname` syscall instead of `gethostname`)
- https://askubuntu.com/questions/294581/whats-the-difference-between-hostname-and-uname-n
- https://golang.org/src/os/sys_windows.go (uses `GetComputerNameEx`, not `GetComputerNameA` like the windows code from Dave Lancaster)

### CODE REVIEW

**Comment by kevinkassimo** [comment]
I think if we are concerned about dependency we can probably just copy the code and maintain ourselves in the future (since it is a very small crate)

**Comment by crabmusket** [comment]
Am I right in thinking `third_party` is the place to vendor dependencies? I'll try to get that working with the `lunaryorn` crate

File: cli/ops.rs [hunk 1]
```diff
@@ -2216,6 +2218,34 @@ fn op_host_get_message(
   Ok(Op::Async(Box::new(op)))
 }
 
+fn op_hostname(
+  _state: &ThreadSafeState,
+  base: &msg::Base<'_>,
+  data: Option<PinnedBuf>,
+) -> CliOpResult {
+  assert!(data.is_none());
+  let cmd_id = base.cmd_id();
```
Comment (ry) [inline cli/ops.rs]: I think we should check for --allow-env here.

**Comment by crabmusket** [comment]
I'm going to redo this to a) implement the new JSON op parsing, b) add a third-party crate. Changes seem too major to keep this current PR.

**Pull request closed without merging**